cli/cluster/config.go

```diff
@@ -0,0 +1,161 @@
+package cluster
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	fmtPathNotMap   = "path %v is not a map"
+	fmtPathNotExist = "path %v does not exist"
+)
+
+// Config is a container for deserialized configuration.
+type Config struct {
+	paths any
+}
+
+// NewConfig creates a new empty configuration.
+func NewConfig() *Config {
+	return &Config{}
+}
+
+func (c *Config) createMaps(path []string) error {
+	var (
+		prev    map[any]any
+		prevKey string
+		cur     any
+	)
+
+	cur = c.paths
+	for i := 0; i < len(path)+1; i++ {
+		if cur == nil {
+			cur = make(map[any]any)
+		} else if _, ok := cur.(map[any]any); !ok {
+			return fmt.Errorf(fmtPathNotMap, path[0:i])
+		}
+
+		if prev == nil {
+			c.paths = cur
+		} else {
+			prev[prevKey] = cur
+		}
+
+		if i < len(path) {
+			prev = cur.(map[any]any)
+			prevKey = path[i]
+			cur = prev[prevKey]
+		}
+	}
+	return nil
+}
+
+func (c *Config) getMap(path []string) (map[any]any, error) {
+	cur := c.paths
+	for i := 0; i < len(path)+1; i++ {
+		if m, ok := cur.(map[any]any); !ok {
+			return nil, fmt.Errorf(fmtPathNotMap, path[0:i])
+		} else if i < len(path) {
+			if _, ok := m[path[i]]; !ok {
+				return nil, fmt.Errorf(fmtPathNotExist, path[0:i+1])
+			} else {
+				cur = m[path[i]]
+			}
+		}
+	}
+
+	return cur.(map[any]any), nil
+}
+
+// Set sets a value to a configuration path.
+func (c *Config) Set(path []string, value any) error {
+	if path == nil || len(path) == 0 {
+		c.paths = value
+		return nil
+	}
+
+	last := len(path) - 1
+	if err := c.createMaps(path[0:last]); err != nil {
```

Comment:
Isn't it useful for `createMaps` to return the last created map?

Reply:
Updated.